roles/openshift_docker_facts/tasks/main.yml

```diff
@@ -37,3 +40,46 @@
 - set_fact:
     docker_options: "{{ openshift.docker.options | default(omit) }}"
   when: not openshift.docker.hosted_registry_insecure | default(False) | bool
+
+# Avoid docker 1.9 when installing origin < 1.2 or OSE < 3.2 on RHEL/Centos and
+# See: https://bugzilla.redhat.com/show_bug.cgi?id=1304038
+- name: Default to latest docker
+  set_fact:
+    docker_version: ''
+    docker_downgrade: False
+  
+- name: Gather installed version of common package
+  shell: >
+    rpm -q {{ openshift.common.service_type }} --queryformat '%{version}' | cut -d '.' -f1,2 | grep -v 'not installed'
+  register: installed_common_version
+  failed_when: false
+  changed_when: false
+  
+- name: Gather available version of common package
+  shell: >
+    yum list available -e 0 -q "{{ openshift.common.service_type}}" 2>&1 | tail -n +2 | awk '{ print $2 }' | sort -r | tr '\n' ' ' | tail -n 1 | cut -d '.' -f1,2
+  register: available_common_version
+  failed_when: false
+  changed_when: false
+
+- name: Gather installed version of docker
+  shell: >
+    rpm -q docker --queryformat '%{version}' | cut -d '.' -f1,2 | grep -v 'not installed'
+  register: installed_docker_version
+  failed_when: false
+  changed_when: false
+  
+- name: Gather available version of docker
+  shell: >
+    yum list available -e 0 -q "docker" 2>&1 | tail -n +2 | awk '{ print $2 }' | sort -r | tr '\n' ' ' | tail -n 1
+  register: available_docker_version
+  failed_when: false
+  changed_when: false
+
+- set_fact:
+    docker_version: "{{ '1.8.2' }}"
+  when: " installed_common_version.stdout in ['3.0','3.1','1.0','1.1'] or available_common_version.stdout in ['3.0','3.1','1.0','1.1']"
```

Comment:
The centos docker-1.9 package unfortunately `Conflicts: origin < 1.2` so I think we should leave it as is.

You mean docker_version and docker_downgrade? Earlier I just set them to default values and then only set them otherwise as needed. I may be misunderstanding your question though.